From a710bb9719f54b2118c523510ef96a6f853e8cab Mon Sep 17 00:00:00 2001
From: Condor3397 <freedomcondor@126.com>
Date: Fri, 16 Mar 2018 11:54:58 +0100
Subject: [PATCH] iss

---
 drivers/staging/media/omap4iss/iss.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/drivers/staging/media/omap4iss/iss.c b/drivers/staging/media/omap4iss/iss.c
index c26c99fd4a24..a48273c884c0 100644
--- a/drivers/staging/media/omap4iss/iss.c
+++ b/drivers/staging/media/omap4iss/iss.c
@@ -1344,12 +1344,21 @@ static const struct platform_device_id omap4iss_id_table[] = {
 };
 MODULE_DEVICE_TABLE(platform, omap4iss_id_table);
 
+/* add of match table  */
+static const struct of_device_id omap4iss_of_table[] = {
+	{ .compatible = "ti,omap4iss" },
+	{ },
+};
+MODULE_DEVICE_TABLE(of, omap4iss_of_table);
+/* end adding of match table  */
+
 static struct platform_driver iss_driver = {
 	.probe		= iss_probe,
 	.remove		= iss_remove,
 	.id_table	= omap4iss_id_table,
 	.driver = {
 		.name	= "omap4iss",
+		.of_match_table = of_match_ptr(omap4iss_of_table),
 	},
 };
 
-- 
2.14.1

