From f53dcc7538a30fbd09488ca1fa7d13f13b721f7b Mon Sep 17 00:00:00 2001
From: Condor3397 <freedomcondor@126.com>
Date: Mon, 30 Apr 2018 14:30:40 +0200
Subject: [PATCH] omap4dtschanged

---
 arch/arm/boot/dts/omap4.dtsi | 39 +++++++++++++++++++++++++++++++++++++++
 1 file changed, 39 insertions(+)

diff --git a/arch/arm/boot/dts/omap4.dtsi b/arch/arm/boot/dts/omap4.dtsi
index 1bcf66ec209f..ce68faad7faf 100644
--- a/arch/arm/boot/dts/omap4.dtsi
+++ b/arch/arm/boot/dts/omap4.dtsi
@@ -573,6 +573,45 @@
 			#iommu-cells = <0>;
 		};
 
+		target-module@52000000 {
+		//omap4iss : omap4iss@52000000 {
+			compatible = "simple-bus";
+			ti,hwmods = "iss";
+
+			/* reg here will just contain rev, sysc and syss eventually */
+			reg = <0x52000000 0x1000000>;
+
+			#address-cells = <1>;
+			#size-cells = <1>;
+			ranges = <0 0x52000000 0x1000000>;
+
+			iss: iss@0 {
+				compatible = "ti,omap4iss";
+				reg =  	<0x0000 0x200>,		/* top */
+						<0x1000 0x170>, /* csi2_a_regs1 */
+						<0x1170 0x020>, /* camerarx_core1 */
+						<0x1400 0x170>, /* csi2_b_regs1 */
+						<0x1570 0x020>, /* camerarx_core2 */
+						<0x2000 0x200>, /* bte */
+						<0x10000 0x0a0>, /* isp_sys1 */
+						<0x10400 0x400>, /* isp_resizer */
+						<0x10800 0x800>, /* isp_ipipe */
+						<0x11000 0x200>, /* isp_isif */
+						<0x11200 0x080>; /* isp_ipipeif */
+
+				harry = <0x123 0x321 0x456>;			// harryflag
+				//syscon = <&scm_conf 0x6c>;	// copy from isp
+				syscon = <&scm_conf>; 
+				//status = "ok";
+				//ti,hwmods = "iss";
+				interrupts = <GIC_SPI 24 IRQ_TYPE_LEVEL_HIGH>;
+				clocks = <&ducati_clk_mux_ck>, <&iss_ctrlclk>;
+				clock-names = "iss_fck", "iss_ctrlclk";
+				dmas = <&sdma 9>, <&sdma 10>, <&sdma 12>, <&sdma 13>;
+				dma-names = "1", "2", "3", "4";	// only 4 originally, 5 is a flag
+			};
+		};
+
 		mmu_ipu: mmu@55082000 {
 			compatible = "ti,omap4-iommu";
 			reg = <0x55082000 0x100>;
-- 
2.14.1

